modules/howtos/pages/json.adoc

```diff
@@ -1 +1,302 @@
+
+If we read the contents of a file to get a JSON string, we can also inflate a JsonObject from that string.
+
+[source,java]
+----
+include::example$Json.java[tags=file,indent=0]
+----
+
+This might seem a little clumsy though:
+We're reading a string, inflating it into a JsonObject, passing it to a serializer which will... turn it back into a string again to send over the network to the Couchbase server!
+
+We could instead simply pass it through with the Raw JSON Transcoder:
+
+[source,java]
+----
+include::example$Json.java[tags=file-raw-imports,indent=0]
+----
+
+[source,java]
+----
+include::example$Json.java[tags=file-raw,indent=0]
+----
+
+This approach could also be used to handle JSON created by other libraries such as Gson.
+See the xref:transcoders-nonjson.adoc#rawjsontranscoder[Transcoder documentation] for more details.
+
+== Serializing objects
+
+It's common to want to serialize your own objects, rather than creating Json programmatically or textually. 
+Luckily our built-in JSON serializer makes light work of this task.
+Using this simple class:
+
+[source,java]
+----
+include::example$Json.java[tags=person-class,indent=0]
+----
+
+We could insert the exact same JSON as before with:
+
+[source,java]
+----
+include::example$Json.java[tags=insert-person,indent=0]
+----
+
+It's now trivial to return the data either as a JsonObject, exactly as we've done before, or indeed as a Person object:
+
+[source,java]
+----
+include::example$Json.java[tags=get-person,indent=0]
+----
+
+[#ObjectMapper]
+== More complex Object to JSON mapping
+
+Because the SDK's serializer is backed by an industry standard library (Jackson), it is easy to model all kinds of Object to JSON mappings.
+Though the SDK automatically uses your version of Jackson when it finds the library on the CLASSPATH, you may wish to be explicit, or use an already configured ObjectMapper, as in this example:
+
+[source,java]
+----
+include::example$Json.java[tags=object-mapper-imports,indent=0]
+----
+
+[source,java]
+----
+include::example$Json.java[tags=object-mapper;!object-mapper-dates,indent=0]
+----
+
+With this setup, we can then configure our classes as we wish.
+Here, we've just renamed the Java class and its fields in Spanish.
+But the generated JSON will have the same fields as before, for easy interoperability.
+
+[source,java]
+----
+include::example$Json.java[tags=persona-class;!exclude,indent=0]
+----
+
+[#datatypes]
+== Modelling JSON datatypes in Java
+
+We've only looked at a few datatypes so far, JSON objects `{}`, strings, and integers. But the full range of JSON types can be expressed, with the obvious mappings to Java.
+
+* string `"hello"`
+* integer `123`
+* float `123.456`
+* boolean: `true`, `false`
+* object: `JavaObject`, `Map`
+* array: `JavaArray`, `List`
+* null
+
+In general, these values can be inserted as top-level values, or nested within an object or array. Let's look at a few examples:
+
+=== Nested object
```

Comment:
Should we use the `:page-toclevels: 2` attribute at the beginning of the doc, so we can jump to these h3s?

Wonder if we should just add that in `attributes.adoc` even?

Reply:
Good idea!
attributes.adoc (or the antora.yml...?) prob needs discussion, as it would change nav for all pages.

Reply:
yeah, let's discuss extracting this outside at the docs sync up maybe, for now I'm fine with it being in this doc 👍 .